### PR TITLE
fix: clarify `--path` is the installation source not destination

### LINKED ISSUE #13200: Cargo install documentation should no longer mention default location
### Problem

cargo install without --path is no longer supported. cargo --list explains cargo install as "Install a Rust binary. Default location is $HOME/.cargo/bin". Please replace "Default location ... " with "Requires --path".

### Steps

cargo --list ; cargo install

### Possible Solution(s)

_No response_

### Notes

_No response_

### Version

cargo 1.74.1 (ecb9851af 2023-10-18)

### PR DESCRIPTION
**Pull request by weihanglo**
### What does this PR try to resolve?

There is a misunderstanding of “default location” mentioned in `cargo install --help`.
I don't think one-line explanation helps much there, So remove that imprecise description.

This PR also add more "from" for places mentioning installation sources.

### How should we test and review this PR?

Read the new help text.

### Additional information

Fixes #13200

### CODE REVIEW

File: src/bin/cargo/cli.rs [hunk 1]
```diff
@@ -569,7 +569,7 @@ Rust's package manager
     <cyan,bold>update</>      Update dependencies listed in Cargo.lock
     <cyan,bold>search</>      Search registry for crates
     <cyan,bold>publish</>     Package and upload this package to the registry
-    <cyan,bold>install</>     Install a Rust binary. Default location is $HOME/.cargo/bin
+    <cyan,bold>install</>     Install a Rust binary.
```
Comment (hi-rustin) [inline src/bin/cargo/cli.rs:572]:
```suggestion
    <cyan,bold>install</>     Install a Rust binary
```

Maybe we should remove this period here to align with other commands.

**Review by hi-rustin — APPROVED** [review]
Looks good to me. Thanks! 👍

**Review by epage — APPROVED** [review]
Feel free to r= once you take care of hi-rustin's comment

**Comment by weihanglo** [comment]
Thank you all :)

@bors r=hi-rustin

**Pull request merged**